src/main/java/opentree/plugins/tree_of_life.java

```diff
@@ -205,49 +206,59 @@ public Representation mrca (@Source GraphDatabaseService graphDb,
+        Map<String, Object> v2_result = new HashMap<>(); // the result we're returning
+        Map<String, Object> v3_result; // the result returned by the v3 method
+        try {
+            v3_result = v3.doMrca(graphDb, longIdsToStringIds(nodeIDs), ottIDs);
+            v2_result.put("node_ids_not_in_tree", empty);
+            v2_result.put("ott_ids_not_in_tree", empty);
+        } catch (BadIdsException e) {
+            v3_result = e.json;
+            v2_result.put("node_ids_not_in_tree", stringIdsToLongIds(e.nodeIds));
+            v2_result.put("ott_ids_not_in_tree", e.ottIds);
+        }
+
+        v2_result.put("invalid_node_ids", empty);
+        v2_result.put("invalid_ott_ids", empty);
+        Map<String, Object> mrca = (Map<String, Object>)v3_result.get("mrca");
+        Map<String, Object> nearest = (Map<String, Object>)v3_result.get("nearest_taxon");
 
-        Map<String, Object> res = new HashMap<>();
         String nodeId = (String)(mrca.get("node_id"));
-        res.put("mrca_node_id", stringIdToLongId(nodeId));
-        res.put("invalid_node_ids", empty);
-        res.put("invalid_ott_ids", empty);
-        res.put("node_ids_not_in_tree", empty);
-        res.put("ott_ids_not_in_tree", empty);
-        res.put("tree_id", treeId);
+        v2_result.put("mrca_node_id", stringIdToLongId(nodeId));
+        v2_result.put("tree_id", treeId);
 
         Map<String, Object> taxon = (Map<String, Object>)mrca.get("taxon");
         if (taxon != null) {
-            res.put("ott_id", taxon.get("ott_id"));
+            v2_result.put("ott_id", taxon.get("ott_id"));
             String name = (String)taxon.get("name");
-            res.put("mrca_name", name);
-            res.put("mrca_rank", taxon.get("rank"));
+            v2_result.put("mrca_name", name);
+            v2_result.put("mrca_rank", taxon.get("rank"));
             String uname = (String)taxon.get("unique_name");
             if (uname.equals(name))
-                res.put("mrca_unique_name", "");
+                v2_result.put("mrca_unique_name", "");
             else
-                res.put("mrca_unique_name", uname);
-            res.put("nearest_taxon_mrca_node_id", nodeId);
+                v2_result.put("mrca_unique_name", uname);
+            v2_result.put("nearest_taxon_mrca_node_id", stringIdToLongId(nodeId));
         } else {
-            res.put("ott_id", "null");
-            res.put("mrca_name", "");
-            res.put("mrca_rank", "");
-            res.put("mrca_unique_name", "");
+            v2_result.put("ott_id", "null");
+            v2_result.put("mrca_name", "");
+            v2_result.put("mrca_rank", "");
+            v2_result.put("mrca_unique_name", "");
         }
-        if (nearest != null) {
+        // Not sure what the old v2 API did in this situation, I think
+        // it just put random values, or no values at all
+        if (nearest == null) nearest = taxon;
+        {
```

Comment:
This block is no longer associated with the `if` test above, so these lines run in al cases. Why not remove the braces + block?